src/monmove.cpp

```diff
@@ -38,6 +40,69 @@ bool monster::wander()
     return ( goal == pos() );
 }
 
+bool monster::is_dangerous_field( const field &fd ) const
+{
+    // No fields = not dangerous (obviously)
+    if( fd.fieldCount() == 0 ) {
+        return false;
+    }
+
+    // Else check each field to see if it's dangerous to us
+    for( auto &fld : fd ) {
+        if( is_dangerous_field( fld.second ) ) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool monster::is_dangerous_field( const field_entry &entry ) const
+{
+    const field_id fid = entry.getFieldType();
+    // Check immunity towards fields
+    switch( fid ) {
+        case fd_smoke:
+        case fd_tear_gas:
+        case fd_toxic_gas:
+        case fd_relax_gas:
+        case fd_nuke_gas:
+            if (has_flag( MF_NO_BREATHE )) {
+                return false;
+            }
+            break;
```

Comment:
You can move the `switch` into `bool monster::is_dangerous_field( const field_id &fid )` and write there:
```c++
...
case fd_toxic_gas:
case fd_relax_gas:
case fd_nuke_gas:
   return !has_flag( MF_NO_BREATHE );
...

```
This would be alot shorter.